src/_guides/language/effective-dart/design.md

```diff
@@ -1103,54 +1093,49 @@ or a `File` object? Is this method synchronous or asynchronous? This is clearer:
-signature, just that you should *annotate* it explicitly when you do so.
+* Literals.
+* Constructor invocations.
+* References to other constants that are explicitly typed.
+* Simple expressions on numbers and strings.
+* Factory methods like `int.parse()`, `Future.wait()`, etc. that readers are
+  expected to be familiar with.
 
+In general, err on the side of annotating. You may still wish to explicitly
+annotate even the type is obvious. If the inferred type relies on values or
+declarations from other libraries, you may want to annotate the type in *your*
+declaration so that a change to that other library doesn't silently change the
+type of your own API without you realizing.
 
-### PREFER type annotating private declarations whose type isn't inferred.
 
-Type annotations on your public declarations help *users* of your code. Nearly
-as important is guiding *maintainers* of your code. Adding type annotations to
-private member and variable declarations can help future readers of your code
-understand it, and help corral bugs.
+### CONSIDER type annotating private top-level variables and fields if the type isn't obvious.
 
-{:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (type-private)"?>
-{% prettify dart %}
-class CallChainVisitor {
-  final SourceVisitor _visitor;
-  final Expression _target;
-  // ···
-  void _writeCall(Expression call) { ... }
-}
-{% endprettify %}
+Type annotations on your public declarations help *users* of your code. Types on
+private members help *maintainers*. The scope of a private declaration is
+smaller and those who need to know the type of that declaration are also more
+likely to be familiar with the surrounding code. That makes it reasonable to
+lean more heavily on inference and omit types in private declarations, which is
+why this guideline is softer than the previous one.
 
-This guideline is deliberately softer than the previous one because the benefits
-of static typing are smaller when code is confined within a single library. In
-practice, though, most Dart users type annotate all declarations, public or
-private.
+If you think the initializer expression&mdash;whatever it is&mdash;is
+sufficiently clear, then you may omit the annotation. But if you think
+annotating helps make the code clear, then by all means add one.
 
 
-### AVOID annotating types for initialized local variables.
+### AVOID type annotating initialized local variables.
 
-Function bodies in idiomatic code tend to be short, and the types of local
-variables are almost always obvious and inferrable from the initializing
-expression. Writing that type down is often simply visual noise. Instead, spend
-that energy choosing a great *name* for your local variable.
+Local variables, especially in modern code where functions tend to be small,
+have very little scope. Omitting the type focuses the reader's attention on the
+more important *name* of the variable and its initialized value.
```

Comment:
Why do the two examples have different local variable names? I found myself distracted by that, because I expected only the type annotations to be different.

Reply:
The idea, which probably is lost on the reader, was that by not worrying about the type annotation, the author had the time to choose a better name. Probably trying to be too clever. Changed to use the same names.